--- netrik-1.2.1/configure.in~	Sun Jan 26 16:49:17 2003
+++ netrik-1.2.1/configure.in	Sun Jan 26 16:49:17 2003
@@ -17,11 +17,7 @@
   CFLAGS=""
 fi;
 
-CFLAGS="$CFLAGS -g -W -Wall -Wstrict-prototypes -Wmissing-prototypes"
-
-AC_ARG_ENABLE(optimization,
-[  --enable-optimization   use compiler optimization],
-CFLAGS="$CFLAGS -O3 -fforce-addr -fforce-mem", CFLAGS="$CFLAGS")
+CFLAGS="$CFLAGS -W -Wall -Wstrict-prototypes -Wmissing-prototypes"
 
 dnl Checks for programs.
 AC_PROG_CC
