--- netrik-1.16.1/configure.ac.orig	2009-01-02 03:46:45.000000000 +0100
+++ netrik-1.16.1/configure.ac	2011-02-20 12:23:08.830975068 +0100
@@ -19,11 +19,11 @@
 
 dnl check for ncurses
 AC_CHECK_LIB(ncursesw,
-	     tparm,
+	     initscr,
 	     [NCURSES="yes"],
 	     AC_MSG_ERROR([ncursesw is absolutely required!]))
 
-LIBS="-lncursesw $LIBS"
+LIBS="-lncursesw -ltinfow $LIBS"
 
 dnl check for readline
 AC_CHECK_LIB(readline,
